### PR TITLE
Release lock before .WaitUntil

### PR DESCRIPTION
**Pull request by ANPez**
Release the lock before .WaitUntil prevents deadlocks when testing parallel objects.

Address #167: Unable to test concurrent objects

### CODE REVIEW

**Comment by ernesto-jimenez** [comment]
@ANPez could you add a test so we avoid regressions in the future?

**Comment by DAddYE** [comment]
> @ANPez could you add a test so we avoid regressions in the future?

Agree

**Comment by ernesto-jimenez** [comment]
@ANPez small nudge, the tests from this PR are not passing :)

**Comment by ANPez** [comment]
Fixed it! Looks like since I wrote it, some code around it changed :)

File: mock/mock_test.go [hunk 1]
```diff
@@ -987,3 +987,29 @@ func Test_Arguments_Bool(t *testing.T) {
...
+	// Lock both goroutines on the .WaitUntil method
+	go func() {
+		mockedService.TheExampleMethod2(false)
+	}()
+	go func() {
+		mockedService.TheExampleMethod2(true)
+	}()
+
+	// Allow the first call to execute, so the second one executes afterwards
+	ch2 <- time.Now()
```
Comment (ernesto-jimenez) [inline mock/mock_test.go]: Maybe it would be good to have each `go func()` write something and assert the order is correct at the end of the test?

**Pull request merged**